shared/src/native-src/pal.h

```diff
@@ -39,7 +39,38 @@ struct has_deprecated_log_folder<T, decltype(T::logging_environment::deprecated_
 };
 
 template <class TLoggerPolicy>
-inline shared::WSTRING GetDatadogLogFilePath(const std::string& file_name_suffix)
+inline fs::path GetDefaultLogDir()
+{
+    bool isAas = false;
+    TryParseBooleanEnvironmentValue(GetEnvironmentValue(WStr("DD_AZURE_APP_SERVICES")), isAas);
+
+    if (isAas)
+    {
+#ifdef _WINDOWS
+        return WStr("C:\\home\\LogFiles\\datadog\\");
+#else
+        return WStr("/home/LogFiles/datadog/");
+#endif
+    }
```

Comment:
@andrewlock 
I told to serverless team and they recommended this
I would like to have your input and be sure that it's aligned with the managed part of the tracer

Reply:
On windows, we set the log dir [here](https://github.com/DataDog/datadog-aas-extension/blob/master/dotnet/content/applicationHost.xdt#L64C9-L64C142). It allows to have log per version. Can't you use that?

Reply:
We use it already for windows and there is nothing for Linux. My PR was mainly for Linux and I was wondering if I should care of Windows too

Reply:
As discussed offline, I think these changes make sense
- We already set the log directory in the AAS extension (Windows) so none of these changes will affect the non-consumption plans
- Default to those special paths on _all_ AAS. Currently we only use these defaults on consumption plans, but there's no reason we should do that.
- Use the suggested paths above - we're currently using different ones on the managed side